src/Adapter/Language/ContextLanguageDataProvider.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Adapter\Language;
+
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+
+/**
+ * Class ContextLanguageDataProvider is responsible for getting languages related data from legacy content class
+ */
+class ContextLanguageDataProvider
```

Comment:
I thought at first but it's strictly related with the legacy context so moving to LanguageDataProvider might confuse someone - decided to name adapter so it would be clear that the language retrieved from it is related with the shops context.